### DISCUSSION TITLE
mesh_optimizer workflows?

### DISCUSSION THREAD

**Discussion by wnissen** [mfem/mfem#3392] [General]
Hi, I am trying to do a primitive parameter study with `mesh_optimizer`. An example run would be:
```
${MFEM_DIR}/miniapps/meshing/mesh-optimizer -m cubitTet.genesis --order 3 --refine-serial 0 --lin-solver 3 --metric-id 321 --target-id 2 -nor -fix-bnd
```
I would like to do a couple things. First, I'd like to get an MFEM `.mesh` formatted file of the initial CUBIT Genesis/NetCDF file, just for comparison. Then I'd like to be able to control the output filename rather than having it default to `optimized.mesh`. Any help is appreciated, thanks!

**Comment by mstees** [comment]
@vladotomov might have some suggestions on this.

To get the initial mesh in mfem format, you could comment out lines 411-413 in mesh-optimizer.cpp:
![image](https://user-images.githubusercontent.com/381496/210884705-0c2a558c-4236-4de6-845f-42f39c82ff00.png)
The initial mesh should then be written to "perturbed.mesh".

My first thought for changing the output name is to add an additional flag with your desired string name and update line 1206 in mesh-optimizer.cpp to reflect that new string.
![image](https://user-images.githubusercontent.com/381496/210885568-d4d2e214-993c-40f7-8c83-fc28ce02e5d9.png)

**Reply by wnissen** [reply]
Thanks, I think I got the original mesh by setting the number of newton iterations to 0 and not increasing the order on input, I will check if that does the same. E.g.:
```
--order 2 --newton-iters 0 --lin-iter 0
```
I think I may run this in Maestro so I'll have a different directory for each run, so then I would have full metadata instead of the filename, that may be better anyway.
Thanks for your help!

**Reply by vladotomov** [reply]
Yes as Mike said, the initial mesh is saved in `perturbed.mesh`, and you can create a new command-line parameter in the beginning and then replace the default:
```
const char *file_name = "optimized.mesh";
...
args.AddOption(&file_name, "-file", "--filename", "Output file name");
...
ofstream mesh_ofs(file_name);
```
Let us know if we can help.

**Reply by wnissen** [reply]
Ah, that's true, it would be easy to add, thanks.
